src/main.ts

```diff
@@ -28,17 +32,16 @@ async function run(): Promise<void> {
     core.exportVariable('GOOGLE_APPLICATION_CREDENTIALS', serviceAccountFile)
 
     core.debug('Authenticating: creating auth client')
-    const auth = new google.auth.GoogleAuth({
+    const authClient = new auth.GoogleAuth({
```

Comment:
this is never used - I guess `auth.GoogleAuth` manages this internally somehow and just passing `auth` to future methods "just works"?

Reply:
I haven't verified it yet, but we get type mismatch issues if we pass in the result of getAuthClient, whereas this seems to be ok.

It's strange, considering the documentation for them doesn't reflect this either.

Reply:
I double-checked the type def and it definitely is set up to accept a GoogleAuth directly. I say we merge it and hope for the best 😆